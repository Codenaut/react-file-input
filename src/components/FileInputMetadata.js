import React from 'react';
import { string, number } from 'prop-types';
import '../styles/FileInputMetadata.scss';

const FileInputMetadata = props => <div className="brainhubFileInput__metadata">
  <div className="brainhubFileInput__metadata__name">{props.name}</div> ({props.size})
</div>;

FileInputMetadata.propTypes = {
  name: string.isRequired,
  size: number.isRequired,
};

export default FileInputMetadata;

