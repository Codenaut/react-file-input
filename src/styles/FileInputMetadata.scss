.brainhubFileInput__metadata {
  background: rgb(137, 137, 214);
  // display: grid;
  // grid-template-columns: repeat(2 1fr);
  // grid-template-rows: 50px;
  // grid-template-areas:
  //   "fn"
  //   "s";
  width: 400px;
  height: 100px;

  .brainhubFileInput__metadata__name {
      font-weight: bold;
  }
  .brainhubFileInput__metadata__size {
      grid-area: s;
      text-align: center;
  }
}
