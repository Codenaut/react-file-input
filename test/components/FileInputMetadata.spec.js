import React from 'react';
import renderer from 'react-test-renderer';

import FileInputMetadata from '../../src/components/FileInputMetadata';

describe('components', () => {
  describe('FileInputMetadata', () => {
    it('should match exact snapshot', () => {
      const tree = renderer.create(
        <div>
          <FileInputMetadata name="test" size={1000}/>
        </div>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
