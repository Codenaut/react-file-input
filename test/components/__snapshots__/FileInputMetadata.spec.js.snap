// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components FileInputMetadata should match exact snapshot 1`] = `
<div>
  <div
    className="brainhub-file-input__metadata"
  >
    <div
      className="brainhub-file-input__metadata__image"
    >
      <div
        style={
          Object {
            "backgroundColor": "#5680BB",
            "height": "20px",
            "width": "20px",
          }
        }
      />
    </div>
    <div
      className="brainhub-file-input__metadata__info"
    >
      <span
        className="brainhub-file-input__metadata__info_name"
      >
        test
      </span>
      
       (
      1000 B
      ) 
      
    </div>
  </div>
</div>
`;
